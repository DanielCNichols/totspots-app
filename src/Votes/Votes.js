import React, { useState, useEffect } from 'react';
import ApiService from '../services/api-service';
import VenuesContext from '../VenuesContext';
import './Votes.css';
import TokenService from '../services/TokenService';
import { withRouter } from 'react-router-dom';
import { TiArrowUpOutline, TiArrowDownOutline } from 'react-icons/ti';

const Votes = ({ count }) => {
  return (
    <div>
      <TiArrowUpOutline />
      <span>{count}</span>
      <TiArrowDownOutline />
    </div>
  );
};

// class Votes extends React.Component {
//   static contextType = VenuesContext;

//   //Allows for accurate vote count with brand new reviews.
//   state = {
//     count: !this.props.review.count
//       ? (this.props.review.count = 0)
//       : parseInt(this.props.review.count),
//     clicked: false,
//   };

//   updateVoteCount() {
//     this.setState({
//       count: this.state.count + 1,
//       clicked: true,
//     });
//   }

//   handleVote = ev => {
//     if (!TokenService.hasToken() === true) {
//       this.props.history.push('/login');
//     } else {
//       let vote = ev.target.value;
//       let id = this.props.review.id;
//       ApiService.handleVotes(vote, id)
//         .then(vote => {
//           this.updateVoteCount(vote);
//         })
//         .catch(error => {
//           this.context.setError(error);
//         });
//     }
//   };

//   renderVote() {
//     const { count } = this.state;
//     if (!count) {
//       return <span>0 people like this review</span>;
//     } else if (count === '1') {
//       return <span>1 person likes this review</span>;
//     }
//     return <span>{count} people like this review</span>;
//   }

//   render() {
//     return (
//       <div className="votes">
//         <p>Was this review helpful?</p>
//         {this.renderVote()}
//         <button
//           disabled={this.state.clicked}
//           value="true"
//           onClick={this.handleVote}
//         >
//           Like
//         </button>
//       </div>
//     );
//   }
// }

export default withRouter(Votes);
