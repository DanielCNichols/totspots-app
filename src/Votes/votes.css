.votes {
  text-align: center;
  margin: 20px 0  30px 0;
}

.votes button {
  display: block;
  min-width: 140px;
  height: 40px;
  margin: 20px auto;
  border-radius: 5px;
  border-color: #ED254E;
  cursor: pointer;
  background-color: transparent;
  font-size: 1.1em;
}

