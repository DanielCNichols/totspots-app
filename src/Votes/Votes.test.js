
import React from 'react';
import ReactDOM from 'react-dom'
import Votes from './Votes.js'
import {MemoryRouter} from 'react-router-dom'
import renderer from 'react-test-renderer'



it('renders Votes without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
  <MemoryRouter>
    <Votes />
  </MemoryRouter>, div);
  ReactDOM.unmountComponentAtNode(div)
})

it('renders Votes as expected', () => {
  const tree = renderer
  .create(
  <MemoryRouter>
    <Votes name="Votes"/>
  </MemoryRouter>
  )
  .toJSON();
  expect(tree).toMatchSnapshot();
})
