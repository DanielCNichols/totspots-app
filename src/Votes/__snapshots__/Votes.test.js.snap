// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Votes as expected 1`] = `
<div
  className="votes"
>
  <p>
    Was this review helpful?
  </p>
  <span>
    (VoteCount) liked this
  </span>
  <button
    onClick={[Function]}
    value="true"
  >
    Like
  </button>
</div>
`;
